core/templates/pages/contributor-dashboard-page/services/contribution-opportunities-backend-api.service.ts

```diff
@@ -247,6 +253,43 @@ export class ContributionOpportunitiesBackendApiService {
       return [];
     }
   }
+
+  async savePreferredTranslationLanguageAsync(
+      languageCode: string
+  ): Promise<void> {
+    return this.userService.getUserInfoAsync().then(
+      (userInfo) => {
+        if (userInfo?.isLoggedIn()) {
+          return this.http.post<void>(
+            '/preferredtranslationlanguage',
+            {language_code: languageCode}
+          ).toPromise().catch((errorResponse) => {
+            throw new Error(errorResponse.error.error);
+          });
+        }
+      }
+    );
+  }
+
+  async getPreferredTranslationLanguageAsync(
+  ): Promise<string> {
+    const emptyResponse = {
+      preferred_translation_language_code: ''
+    };
+    return this.userService.getUserInfoAsync().then(
+      async(userInfo) => {
+        if (userInfo?.isLoggedIn()) {
+          const res = await this.http
+            .get<PreferredTranslationLanguageBackendDict>(
+              '/preferredtranslationlanguage').toPromise().catch(
+              () => emptyResponse);
+          return res.preferred_translation_language_code;
+        } else {
+          return '';
```

Comment:
Why we return empty string and not `null`?

Reply:
If we use `null` then we need to add a lot of if-else statements to deal with the typescript errors. Also, I was trying to follow the general trend which was being followed in this file (like returning empty arrays `[]` instead of `null` in `fetchTranslatableTopicNamesAsync` and `fetchFeaturedTranslationLanguagesAsync`).

Reply:
I think `''` is different from `[]`, we usually add stuff to lists but for strings we usually just set them and then work with them. We generally try not to use `''` except for some rare cases.

Reply:
I have replaced the empty string with `null`.